core/controllers/base.py

```diff
@@ -884,7 +944,10 @@ class OppiaMLVMHandler(BaseHandler):
     """Base class for the handlers that communicate with Oppia-ML VM instances.
     """
 
-    def extract_request_message_vm_id_and_signature(self):
+    # Here we use type Any because this method is implemented in a subclasses
```

Comment:
I don't understand this, it just seems incorrect. Presumably this should be labelled as an abstract method or something like that, if subclasses are the issue?

What are the return types of the subclass methods? This should have the same return type as those.

Reply:
> Presumably this should be labelled as an abstract method

Fixed